src/nvim/ui_compositor.c

```diff
@@ -666,3 +672,20 @@ static void ui_comp_grid_resize(UI *ui, Integer grid, Integer width, Integer hei
   }
 }
 
+static void ui_comp_event(UI *ui, char *name, Array args, bool *args_consumed)
```

Comment:
note: using the compositor for this is somewhat a hack but it enables quick prototyping and you could argue it is used primarily for composed ui:s (i e TUI) as multigrid ui:s will intend to implement external widgets themselves.